provider/openrouter/openrouter.go

```diff
@@ -52,13 +54,28 @@ func (p *Provider) ID() (id string) {
 // Models returns which models are available to be queried via this provider.
 func (p *Provider) Models() (models []model.Model, err error) {
 	client := p.client()
-	ms, err := client.ListModels(context.Background())
-	if err != nil {
-		return nil, pkgerrors.WithStack(err)
+
+	var responseModels openai.ModelsList
+	if err := retry.Do( // Query available models with a retry logic cause "openrouter.ai" has failed us in the past.
```

Comment:
Was thinking about that as well... but that requires us to change the API and pass the logger object.
Since this happens only during initialization, I think we can get away without logging. It's a one-time thing, and if it fails we won't even start evaluating. It's not like the LLM queries themselves where connection issues could potentially be unfair.

Reply:
Sounds good